mmv1/third_party/terraform/.teamcity/components/generated/build_components.erb

```diff
@@ -39,39 +39,53 @@ fun BuildFeatures.Golang() {
     })
 }
 
-fun BuildSteps.DownloadTerraformBinary() {
-    // https://releases.hashicorp.com/terraform/0.12.28/terraform_0.12.28_linux_amd64.zip
-    var terraformUrl = "https://releases.hashicorp.com/terraform/%env.TERRAFORM_CORE_VERSION%/terraform_%env.TERRAFORM_CORE_VERSION%_linux_amd64.zip"
+fun BuildSteps.SetGitCommitBuildId() {
     step(ScriptBuildStep {
-        name = "Download Terraform Core v%env.TERRAFORM_CORE_VERSION%.."
-        scriptContent = "mkdir -p tools && wget -O tf.zip %s && unzip tf.zip && mv terraform tools/".format(terraformUrl)
+        name = "Set build id as shortened git commit hash"
+        scriptContent = """
+            #!/bin/bash
+            GIT_HASH=%system.build.vcs.number%
+            GIT_HASH_SHORT=${'$'}{GIT_HASH:0:7}
+            echo "##teamcity[buildNumber '${'$'}{GIT_HASH_SHORT}']"
+        """.trimIndent()
+        // ${'$'} is required to allow creating a script in TeamCity that contains
+        // parts like ${GIT_HASH_SHORT} without having Kotlin syntax issues. For more info see:
+        // https://youtrack.jetbrains.com/issue/KT-2425/Provide-a-way-for-escaping-the-dollar-sign-symbol-in-multiline-strings-and-string-templates
     })
 }
 
-fun servicePath(path : String, packageName: String) : String {
-    return "./%s/%s".format(path, packageName)
-}
-
-fun BuildSteps.RunAcceptanceTests(path : String, packageName: String) {
-    var packagePath = servicePath(path, packageName)
-    var withTestsDirectoryPath = "##teamcity[setParameter name='PACKAGE_PATH' value='%s/tests']".format(packagePath)
-
-    // some packages use a ./tests folder, others don't - conditionally append that if needed
+fun BuildSteps.DownloadTerraformBinary() {
+    // https://releases.hashicorp.com/terraform/0.12.28/terraform_0.12.28_linux_amd64.zip
+    var terraformUrl = "https://releases.hashicorp.com/terraform/%env.TERRAFORM_CORE_VERSION%/terraform_%env.TERRAFORM_CORE_VERSION%_linux_amd64.zip"
     step(ScriptBuildStep {
-        name          = "Determine Working Directory for this Package"
-        scriptContent = "if [ -d \"%s/tests\" ]; then echo \"%s\"; fi".format(packagePath, withTestsDirectoryPath)
+        name = "Download Terraform version %s".format(defaultTerraformCoreVersion)
+        scriptContent = """
+        #!/bin/bash
+        mkdir -p tools
+        wget -O tf.zip %s
+        unzip tf.zip
+        mv terraform tools/
+        """.format(terraformUrl).trimIndent()
     })
+}
 
+// RunSweepers runs sweepers, and relies on set build configuration parameters
+fun BuildSteps.RunSweepers(sweeperStepName : String) {
     step(ScriptBuildStep{
-        name = "Pre-Sweeper"
-        scriptContent = "go test -v \"%PACKAGE_PATH%\" -sweep=\"%SWEEPER_REGIONS%\"  -sweep-allow-failures -sweep-run=\"%SWEEP_RUN%\" -timeout 30m"
+        name = sweeperStepName
+        scriptContent = "go test -v \"%PACKAGE_PATH%\" -sweep=\"%SWEEPER_REGIONS%\" -sweep-allow-failures -sweep-run=\"%SWEEP_RUN%\" -timeout 30m"
```

Comment:
`gcp_sweeper_test.go.erb` is still in the google package. The command to run the sweeper could be `go test -v "./google" -sweep="us-central1" -sweep-allow-failures -sweep-run="" -timeout 30m`. 

But the command line in the step [Pre-sweeper](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_Google_SarahTest_GOOGLE_PRE_SWEEPER_TESTS/6264?buildTab=log&focusLine=551&linesState=549&logView=flowAware) is `go test -v "./google/sweeper" -sweep="us-central1" -sweep-allow-failures -sweep-run="" -timeout 30m` and it is working well.   I wonder if I miss something?

Reply:
Ah - it's because a [previous version of the code](https://github.com/hashicorp/terraform-provider-google/commit/cb2af7876b071ebf8b6b0a94583bc2ef41b80c27) in my test branch (before my refactor) required the sweeper to be migrated to work correctly.

I think [cb30032](https://github.com/GoogleCloudPlatform/magic-modules/pull/8560/commits/cb3003253bbe58b3bb014111df7eb662f7ff169f) will make this config correct for immediate use, but your PR that moves the sweepers would need to update packages.kt with the new path for the sweeper_test package

Reply:
Thank you for spotting this!

Reply:
Thanks for testing the sweepers migration. I will make the changes to update packages.kt with my PR.